File: ruby/darts/darts.rb
# frozen_string_literal: true

# Implementation of the Darts exercise in Ruby track on Exercism.
class Darts
  def initialize(x, y)
    @x = x.abs
    @y = y.abs

    @boundary = @x * @x + @y * @y
  end

  def score
    return 0 if @boundary > 100
    return 1 if @boundary > 25
    return 5 if @boundary > 1

    10
  end
end

File: lua/darts/darts.lua
local Darts = {}

function Darts.score(x, y)
  boundary = x * x + y * y

  if(boundary > 100)
  then
    return 0
  end
  if(boundary > 25)
  then
    return 1
  end
  if(boundary > 1)
  then
    return 5
  end

  return 10
end

return Darts
